core/src/main/java/org/mule/runtime/core/api/policy/PolicyProvider.java

```diff
@@ -29,6 +29,22 @@
    */
   List<Policy> findSourceParameterizedPolicies(PolicyPointcutParameters policyPointcutParameters);
 
+  /**
+   * @return {@code true} if there is at least one source policy applied that relies on the headers to check for the pointcut
+   *         match.
+   *
+   * @since 4.3
+   */
+  boolean isAnySourcePolicyHeadersAware();
```

Comment:
This is coupling HTTP to the policy engine. It should probably be in http-policy-api

Reply:
yep, fixed